yjit/src/backend/x86_64/mod.rs

```diff
@@ -98,7 +98,7 @@ impl Assembler
             // Load heap object operands into registers because most
             // instructions can't directly work with 64-bit constants
             let opnds = match op {
-                Op::Load | Op::Mov => opnds,
```

Comment:
Here I'm not sure why you removed the mov

Reply:
It's to make sure that we also track GC'ed object when it's a `Mov(mem, value)`.
We only look for values in operands to `Op::Load` in the final emit:

https://github.com/Shopify/ruby/blob/9812895b6ce1dd8b2f6715d7b04af902a8164486/yjit/src/backend/x86_64/mod.rs#L294

Let me know if the comment I added in [01d0cf1](https://github.com/Shopify/ruby/pull/354/commits/01d0cf199c920135206c760e04e79a26b2e7eb33) makes it more clear :)

Reply:
The bit that's confusing to me is that you removed Op::Mov from the match case here.

Is `mov(Mem | Reg, Value)` getting split into `mov(Mem | Reg, load(Value))` somewhere else?